### PR TITLE
Manager e2e testing

### PR DESCRIPTION
**Pull request by Dhruv-J**
New Theia Manager e2e Tests

This change adds end to end testing for the theia manager and cli integration.

Signed-off-by: Dhruv Jain <dhruvj@vmware.com>

### CODE REVIEW

File: test/e2e/policyrecommendation_test.go [hunk 1]
```diff
@@ -174,7 +175,7 @@ func testPolicyRecommendationRun(t *testing.T, data *TestData) {
 	stdout, jobId, err := runJob(t, data)
```
Comment (yuntanghsu) [inline test/e2e/policyrecommendation_test.go:175]:
```suggestion
	stdout, jobName, err := runJob(t, data)
```
Same for other jobID

File: test/e2e/policyrecommendation_test.go [hunk 2]
```diff
@@ -251,7 +252,7 @@ func testPolicyRecommendationFailed(t *testing.T, data *TestData) {
 	})
 	require.NoError(t, err)
 	assert := assert.New(t)
-	assert.Truef(strings.Contains(stdout, "Error message: driver pod not found") || strings.Contains(stdout, "Error message: driver container failed") || strings.Contains(stdout, "Error message: driver container status missing"), "stdout: %s", stdout)
+	assert.Truef(strings.Contains(stdout, "error message: driver pod not found") || strings.Contains(stdout, "error message: driver container failed") || strings.Contains(stdout, "Error message: driver container status missing"), "stdout: %s", stdout)
```
Comment (yuntanghsu) [inline test/e2e/policyrecommendation_test.go:255]:
```suggestion
	assert.Truef(strings.Contains(stdout, "error message: driver pod not found") || strings.Contains(stdout, "error message: driver container failed") || strings.Contains(stdout, "error message: driver container status missing"), "stdout: %s", stdout)
```

File: pkg/controller/networkpolicyrecommendation/controller.go [hunk 3]
```diff
@@ -598,7 +597,7 @@ func (c *NPRecommendationController) startSparkApplication(npReco *crdv1alpha1.N
 	}
 	sparkResourceArgs.executorMemory = npReco.Spec.ExecutorMemory
 
-	recommendationID := uuid.New().String()
+	recommendationID := npReco.Name[3:]
```
Comment (yuntanghsu) [inline pkg/controller/networkpolicyrecommendation/controller.go:600]: Maybe you want to check it is valid npReco.Name?
Like,
`err = commands.ParseRecommendationName(npReco.Name)
	if err != nil {
		return IlleagelArguementError{fmt.Errorf("invalid request: Job name is not a valid policy recommendation job name")}
	}`

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -106,3 +106,34 @@ jobs:
...
+    name: Check whether theia-manager image needs to be built based on diff
+    runs-on: [ubuntu-latest]
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - uses: antrea-io/has-changes@v2
+        id: check_diff
+        with:
+          paths: pkg/* build/images/Dockerfile.theia-manager
```
Comment (yanjunz97) [inline .github/workflows/build.yml:120]:
```suggestion
          paths: pkg/* build/images/Dockerfile.theia-manager.ubuntu
```

File: test/e2e/policyrecommendation_test.go [hunk 5]
```diff
@@ -172,51 +171,51 @@ func TestPolicyRecommendation(t *testing.T) {
 
 // Example output: Successfully created policy recommendation job with ID e998433e-accb-4888-9fc8-06563f073e86
```
Comment (dreamtalen) [inline test/e2e/policyrecommendation_test.go:172]: `ID` -> `name`
Names should have prefix `pr-`?

File: test/e2e/policyrecommendation_test.go [hunk 6]
```diff
@@ -172,51 +171,51 @@ func TestPolicyRecommendation(t *testing.T) {
...
 	require.NoError(t, err)
 	assert := assert.New(t)
 	assert.Containsf(stdout, "Status of this policy recommendation job is", "stdout: %s", stdout)
 }
 
 // Example output:
-// CreationTime          CompletionTime        ID                                   Status
-// 2022-06-17 15:03:24 N/A                 615026a0-1856-4107-87d9-08f7d69819ae RUNNING
+// CreationTime          CompletionTime        Name                                   Status
+// 2022-06-17 15:03:24 N/A                 615026a0-1856-4107-87d9-08f7d69819ae RUNNIN
```
Comment (dreamtalen) [inline test/e2e/policyrecommendation_test.go:192]: Ditto, missing `pr-` prefix of Name column

File: test/e2e/policyrecommendation_test.go [hunk 7]
```diff
@@ -172,51 +171,51 @@ func TestPolicyRecommendation(t *testing.T) {
...
 	assert.Containsf(stdout, "CreationTime", "stdout: %s", stdout)
 	assert.Containsf(stdout, "CompletionTime", "stdout: %s", stdout)
-	assert.Containsf(stdout, "ID", "stdout: %s", stdout)
+	assert.Containsf(stdout, "Name", "stdout: %s", stdout)
 	assert.Containsf(stdout, "Status", "stdout: %s", stdout)
-	assert.Containsf(stdout, jobId, "stdout: %s", stdout)
+	assert.Containsf(stdout, jobName, "stdout: %s", stdout)
 }
 
 // Example output: Successfully deleted policy recommendation job with ID e998433e-accb-4888-9fc8-06563f073e86
```
Comment (dreamtalen) [inline test/e2e/policyrecommendation_test.go:208]: Ditto

File: test/e2e/policyrecommendation_test.go [hunk 8]
```diff
@@ -225,10 +224,10 @@ func testPolicyRecommendationDelete(t *testing.T, data *TestData) {
 // Or
 // Error message: driver container failed
```
Comment (dreamtalen) [inline test/e2e/policyrecommendation_test.go:225]: `Error` -> `error`

**Review by dreamtalen — COMMENTED** [review]
Thanks for the work, overall LGTM with few nits.

File: pkg/controller/networkpolicyrecommendation/controller.go [hunk 9]
```diff
@@ -598,15 +598,19 @@ func (c *NPRecommendationController) startSparkApplication(npReco *crdv1alpha1.N
 	}
 	sparkResourceArgs.executorMemory = npReco.Spec.ExecutorMemory
 
-	recommendationID := uuid.New().String()
+	err = commands.ParseRecommendationName(npReco.Name)
```
Comment (yanjunz97) [inline pkg/controller/networkpolicyrecommendation/controller.go:601]: I understand that using the same uuid for NPR and related SparkApplication may be good in terms of consistency. But maybe move the `ParseRecommendationName` function to `pkg/util`? It is a little bit strange that the controller has dependency on the CLI package.

File: pkg/controller/networkpolicyrecommendation/controller.go [hunk 10]
```diff
@@ -598,15 +598,19 @@ func (c *NPRecommendationController) startSparkApplication(npReco *crdv1alpha1.N
...
+	if err != nil {
+		return IlleagelArguementError{fmt.Errorf("invalid request: Job name is not a valid policy recommendation job name")}
```
Comment (yanjunz97) [inline pkg/controller/networkpolicyrecommendation/controller.go:603]:
```suggestion
		return IlleagelArguementError{fmt.Errorf("invalid request: Policy recommendation job name is invalid: %s", err)}
```

**Comment by yanjunz97** [comment]
I notice the K8s version is updated to 1.23.5, maybe also replace `master` by `master|control-plane`?
https://github.com/antrea-io/theia/blob/8cef9da12a861d71eb33d3b1af5dd8e7a62407af/ci/jenkins/test-vmc.sh#L40-L41

File: test/e2e/policyrecommendation_test.go (hunk 5 above)
Comment (yuntanghsu) [reply test/e2e/policyrecommendation_test.go]: Fixed.

File: test/e2e/policyrecommendation_test.go (hunk 6 above)
Comment (yuntanghsu) [reply test/e2e/policyrecommendation_test.go]: Fixed.

File: test/e2e/policyrecommendation_test.go (hunk 7 above)
Comment (yuntanghsu) [reply test/e2e/policyrecommendation_test.go]: Fixed.

File: test/e2e/policyrecommendation_test.go (hunk 8 above)
Comment (yuntanghsu) [reply test/e2e/policyrecommendation_test.go]: Fixed.

File: .github/workflows/build.yml (hunk 4 above)
Comment (yuntanghsu) [reply .github/workflows/build.yml]: Fixed.

File: pkg/controller/networkpolicyrecommendation/controller.go (hunk 9 above)
Comment (yuntanghsu) [reply pkg/controller/networkpolicyrecommendation/controller.go]: move it to pkg/util/policyrecommendation

File: pkg/controller/networkpolicyrecommendation/controller.go (hunk 10 above)
Comment (yuntanghsu) [reply pkg/controller/networkpolicyrecommendation/controller.go]: fixed. Thanks

**Review by yanjunz97 — APPROVED** [review]
LGTM

**Review by dreamtalen — APPROVED** [review]
LGTM

**Pull request merged**